content/js1/sprints/3/prep/index.md

```diff
@@ -8,4 +8,634 @@ backlog= 'Module-JS1'
+
+Let's describe the strategy for dealing with an input that is after midday.
+
+Earlier we observed that when the time goes beyond midday then **we can subtract 12 from the hours time to get the new hours for the 12 hour clock time**.
+
+_Before_ writing code, we can define our approach in steps using a flowchart:
+
+Starting with an input like `"23:00"`:
+
+```mermaid
+
+flowchart TD
+
+A[Step 1: Extract the hours from the time '23:00', to get '23'] --> B[Step 2: Convert '23' to a number, to get 23]
+B --> C{Step 3\nCheck: Are the hours greater than 12?}
+C -- true --> D[Step 4: subtract 12 from the hours, to get 11]
+D --> E[Step 5: add 'pm' to the rest of the time, to get '11:00 pm']
+E --> F[Step 6: return the new time]
+```
+
+This approach involves running some **conditionally**. In this case, we're only going to continue doing Steps 4-6 if the condition **hours are greater than 12** is `true`.
+
+### ❓ Conditionally executing code
+
+In programming, we can use an `if` statement to execute some code when a given condition is `true`.  
+In JavaScript, we can write an `if` statement as follows:
+
+```js {title="main.js"}
+if (condition) {
+  // do some code in here
+}
+```
+
+The `if` statement consists of:
+
+1. `if` keyword: this is the start of the `if` statement
+2. `(condition)`: some conditional expression that is wrapped in parentheses.
+3. `{}`: a block statement: any code we want to execute if the condition is true goes inside the curly braces here
+
+We can represent this with a diagram too:
+
+```mermaid
+
+flowchart LR
+    IC{condition}
+    IB[Execute code in body]
+    EXIT([End of if statement])
+    IC --> |true| IB
+    IC -.-> |false| EXIT
+    IB --> EXIT
+```
+
+{{<note type="exercise" title="Exercise 2.1">}}
+Write a program that defines a variable `eleniAge`.
+
+If Eleni is less than 18, the program should print out nothing to the console.
+
+If Eleni is 18 years old or older, the program should print out:
+
+```bash
+Eleni is old enough to drive
+```
+
+{{</note>}}
+
+{{<note type="exercise" title="Exercise 2.2">}}
+
+Modify the program above so that if Eleni is less than 18 the program prints out:
+
+```bash
+Eleni is not old enough to drive
+```
+
+{{</note>}}
+
+### 🧠 Applying new knowledge
+
+So for `formatAs12HourClock` we said part of the strategy for handling `"23:00"` would involve checking that the `hours` value is less than 12. For this purpose, we can use another comparison operator: `>`.
+`>` will check if the value on the left of the operator is less than the value on the right of the operator.
+So `3 > 12` would evaluate to be `false`, as `3` is not greater than `12`.
+
+So provided we have an expression for the hours from the time, we can write a conditional statement as follows:
+
+```js
+if (expressionsForHours < 12) {
+  // do code to format the 12 hours
+}
+```
+
+To complete the logic, we now have to achieve a sub-goal:
+
+> 🎯 Sub-goal: Find an expression for the hours from the `time` input
```

Comment:
```suggestion
> 🎯 Sub-goal: Find an expression for the hours from the `time` input, as a number
```

Reply:
I think this and some of the comments would read a lot cleaner if we changed from looking to find the value, not the expression. Something like:

> Sub-goal: Find the hours value from the `time` input

I think that's also semantically more correct? We're not really interested in the expression itself, just the value that it evaluates to.